enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,483 @@
+
+#### Using the package json representation
+
+##### Introducing a new `represent` subcommand for `opm index`
+
+A new subcommand `represent` will be introduced under `opm index` which will have a subcommand of its own called `packages`.
+
+When `opm index represent packages` is summoned, the json representations of the packages inside the index will be unfurled in a folder with the same name as the index. 
+When only `opm index represent` is summoned for an index, the json representations of the packages will be aggregated to output a json representation of the entire index. 
+
+##### Introducing a new `update` subcommand for `opm index`
+
+A new subcommand `update` will be introduced under `opm index`, which will take a folder containing json representations of packages as it's input along with an existing index image tag, and will push any update made to the representations up to the index container image. 
+
+> Note: For a seamless UX, the following things should be consider during implementation when the `opm index update` command is summoned: 
+> 1. Validate the files inside the folder given as input to the command to ensure they are valid json representation of indexes (and not random files with random content, which could be potential security risks)
+> 2. If the files are valid files, validate that the index image tag given as input to the command is a valid tag, i.e and image for that tag already exists. If not, throw an error. 
+> 3. If the files are valid, and the index image tag exists, take a gitdiff between the present content of the index (i.e the existing json representations) and the content of the folder given as input. Display the diff as an output of the command with the following message 
+> "The following changes will be updated in the index:"
+> If there is no diff between the content in the container image and the content of the local folder, throw an error with the message "No diff found between index and local content to update."
+
+##### Introducing a new `create` subcommand for `opm index`
+
+A new subcommand `create` will be introduced under`opm index`, that will take a folder containing json representation of packages, and a container image tag as input, and will create a new index with the package representations fed to it. 
+Since `opm index add` will be modified to create the representation, the `create` subcommand will only serve as a UX facilitator for users. i.e underneath the hood, the `create` subcommand only needs to iterate over the pacakge representations and call the `opm index add` command with the appropriate `path` and `channel` info. 
+
+>Note: Best effort will be made to adhere to the UX mentioned above however appropriate fot this command too. 
```

Comment:
I have difficult seeing the difference here between `opm index create` and `opm index update`. Can you elaborate on the scenarios where you would use one vs. the other?

Reply:
@dmesser [this](https://github.com/operator-framework/enhancements/pull/37#discussion_r540554765) should clear things up.